src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -422,6 +425,111 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point) thro
+        //   1.  shell orientation is cw and range is greater than a hemisphere (180 degrees) but not spanning 2 hemispheres
+        //       (translation would result in a collapsed poly)
+        //   2.  the shell of the candidate hole has been translated (to preserve the coordinate system)
+        boolean incorrectOrientation = component == 0 && handedness != orientation;
+        if ( translated = ((incorrectOrientation && (rng > DATELINE && rng != 360.0)) || (shellCorrected && component != 0))) {
+            for (GeoPoint c : points) {
+                if (c.x < 0.0) {
+                    c.x += 360.0;
+                }
+            }
+        }
+        return translated;
+    }
+
+    public static boolean computePolyOrientation(GeoPoint[] points) {
+        return computePolyOrientation(points, points.length, 0);
+    }
+
+    public static boolean computePolyOrientation(GeoPoint[] points, int length, int offset) {
+        // calculate the direction of the points:
+        // find the point a the top of the set and check its
+        // neighbors orientation. So direction is equivalent
+        // to clockwise/counterclockwise
+        final int top = computePolyTop(points, offset, length);
+        final int prev = (offset + ((top + length - 1) % length));
+        final int next = (offset + ((top + 1) % length));
+        return (points[offset + prev].x > points[offset + next].x);
+    }
+
+    private static final int computePolyTop(GeoPoint[] points, int offset, int length) {
+        int top = 0; // we start at 1 here since top points to 0
+        for (int i = 1; i < length; i++) {
+            if (points[offset + i].y < points[offset + top].y) {
+                top = i;
+            } else if (points[offset + i].y == points[offset + top].y) {
+                if (points[offset + i].x < points[offset + top].x) {
+                    top = i;
+                }
+            }
+        }
+        return top;
+    }
+
+    public static final Pair computeBBox(GeoPoint[] points) {
+        return computeBBox(points, 0, 0);
+    }
+
+    public static final Pair computeBBox(GeoPoint[] points, int offset, int length) {
+        double minX = points[0].x;
+        double maxX = points[0].x;
+        double minY = points[0].y;
+        double maxY = points[0].y;
+        // compute the bounding coordinates (@todo: cleanup brute force)
+        for (int i = 1; i < length; ++i) {
+            if (points[offset + i].x < minX) {
+                minX = points[offset + i].x;
+            }
+            if (points[offset + i].x > maxX) {
+                maxX = points[offset + i].x;
+            }
+            if (points[offset + i].y < minY) {
+                minY = points[offset + i].y;
+            }
+            if (points[offset + i].y > maxY) {
+                maxY = points[offset + i].y;
+            }
+        }
+        return Pair.of(Pair.of(minX, maxX), Pair.of(minY, maxY));
```

Comment:
It would be good to document what this function is returning. I would have expected a pair of x,y coords, but it seems like you are doing the x's and y's together? I haven't seen this before so wanted to make sure that is what was meant..